test/lib/smart-survey-api.test.js

```diff
@@ -8,15 +8,18 @@ let getResponsesStub = sinon.stub();
 const smartSurveyStub = function() {
     return {getResponses: getResponsesStub};
 };
-// Proxyquire syntax: first argument: point to your module (which holds your dependency)
-// 2nd argument: An object with the key: module name to proxy & the value the object above
+// Proxyquire syntax:
+// first argument: points to your module (which holds your dependency)
+// 2nd argument: An object. The object has a key that is the name of the moduleof the dependency
+// & the value the object above of the proxy
 // So this means smartSurveyStub will now point to this function, i.e. a stub
-const smartSurveyAPI = proxyquire('../../lib/smart-survey-api', { 'smartsurvey-client': smartSurveyStub});
+const SmartSurveyAPIBase = proxyquire('../../lib/smart-survey-api', { 'smartsurvey-client': smartSurveyStub});
+const smartSurveyAPI = new SmartSurveyAPIBase('mytoken', 'mytokensecret');
 
 describe('smartSurveyAPI', () => {
   describe('getData()', () => {
     it('is a function', () => (typeof smartSurveyAPI.getData).should.equal('function'));
-    it('takes 3 arguments', () => (smartSurveyAPI.getData).should.have.lengthOf(3));
+    // it('takes 2 arguments but 1 default', () => (smartSurveyAPI.getData).should.have.lengthOf(2));
```

Comment:
@daniel-ac-martin @danielepolencic this is sad but function.length does not work how I'd expect.  Apparently,  it only includes parameters before the first one with a default value.   https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Function/length

this is how my function is set up `getData(surveyID, pageOptions = {})`.  As I have one default value then it only returns sadly one rather than two 😞